src/main/java/ch/njol/skript/expressions/ExprTarget.java

```diff
@@ -91,15 +96,42 @@ public Entity convert(LivingEntity en) {
 	}
 
 	@Override
-	public Class<? extends Entity> getReturnType() {
-		return type != null ? type.getType() : Entity.class;
+	@Nullable
+	public Class<?>[] acceptChange(ChangeMode mode) {
+		switch (mode) {
+			case SET:
+			case RESET:
+			case DELETE:
+				return CollectionUtils.array(LivingEntity.class);
+		}
+		return super.acceptChange(mode);
 	}
 
 	@Override
-	public String toString(@Nullable Event e, boolean debug) {
-		if (e == null)
-			return "the target" + (type == null ? "" : "ed " + type) + (getExpr().isDefault() ? "" : " of " + getExpr().toString(e, debug));
-		return Classes.getDebugMessage(getAll(e));
+	public void change(Event e, @Nullable Object[] delta, ChangeMode mode) {
+		switch (mode) {
+			case SET:
+			case RESET:
+			case DELETE:
+				// null will make the entity target-less (reset target) but for players it will remove them
+				LivingEntity target = delta == null ? null : (LivingEntity) delta[0];
+				for (LivingEntity entity : getExpr().getArray(e)) {
+					if (getTime() >= 0 && e instanceof EntityTargetEvent && entity.equals(((EntityTargetEvent) e).getEntity()) && !Delay.isDelayed(e)) {
```

Comment:
I know this isn't your doing, but since the past state is never used. The `setTime` method of this ExprTarget class should return false if the time is -1. This will allow Skript to error properly when the user attempts to set the previous target.